File: Algorithm/Baekjoon/javascript/1019.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split(' ');
const first = +input[0];
const first = 19;
const result = Array.from({length: 10}, (i) => 0)

const calc = (start, end, ten) => {
  count = ~~(start / 10) - ~~(end / 10) + 1
  result.forEach((el, index) => {
    result[index] += count * ten;
  })
}

const increase_by_num = (num, ten) => {
  while(num) {
    result[num % 10] += ten;
    num /= 10;
  }
}

const solution = (start, end, ten) => {
  while(start % 10 !== 0 && start <= end) {
    increase_by_num(start, ten);
    start += 1;
  }

  if (start > end) {
    return;
  }

  while (end % 10 !== 9 && end >= start) {
    increase_by_num(end, ten);
    end -= 1;
  }

  calc(start, end, ten);
  solution(~~(start / 10), ~~(end / 10), ten * 10)
}

solution(1, first, 1);
for(let i of result) {
  console.log(i)
}

File: Algorithm/Baekjoon/python/1019.py
def calc(start, end, ten) :
  count = end // 10 - start // 10 + 1
  for i in range(len(result)):
    result[i] += count * ten

def increase_by_number(num, ten) :
  k = list(str(num))
  for i in k :
    result[int(i)] += ten
  
def solution(start, end, ten) :
  while(start % 10 != 0 and start <= end) :
    increase_by_number(start, ten)
    start += 1

  if start > end:
    return

  while(end % 10 != 9 and end >= start) :
    increase_by_number(end, ten)
    end -= 1
  calc(start, end, ten)
  solution(start // 10, end // 10, ten * 10)

n = int(input())
result = [0 for _ in range(10)];
solution(1, n, 1)
print(*result)
